src/System.CommandLine.Tests/Binding/SetHandlerTests.cs

```diff
@@ -160,6 +161,62 @@ public void If_service_is_not_found_then_an_error_results()
             exitCode.Should().Be(1);
         }
 
+        [Fact]
+        public void If_no_symbol_was_passed_for_binding_then_the_error_message_suggests_a_fix_for_the_first_missing_symbol()
+        {
+            var boolOption = new Option<bool>("-o");
+            var stringArg = new Argument<string>("value");
+
+            var subcommand = new Command("TheCommand")
+            {
+                boolOption,
+                stringArg
+            };
+
+            var command = new RootCommand
+            {
+                subcommand
+            };
+
+            subcommand.SetHandler((bool boolValue, string stringValue) => { });
+
+            var console = new TestConsole();
+
+            command.Invoke("TheCommand -o hi", console);
+
+            console.Error.ToString().Should()
+                   .Contain(
+                       $"No binding target was provided to the handler for command 'TheCommand' for the parameter at position 0. Did you mean to pass one of these?{NewLine}Option<Boolean> -o");
```

Comment:
Brilliant that we are offering a suggestion. And it looks like pretty cool magic making that happen. But a few things:

* There are two options/args available and only one is listed in the message. Can we do both.
* I'd like different language. Can we do: 

$"SetHandler for command 'TheCommand' is missing options or arguments. Did you mean to pass one of these?{NewLine}Option<Boolean> -o");

Reply:
> There are two options/args available and only one is listed in the message. Can we do both.

They're filtered by type.